docs/changelog/110901.yaml

```diff
@@ -0,0 +1,5 @@
+pr: 110901
+summary: Gracefully handle invalid synonym rules in synonyms sets
```

Comment:
Nitpick: Do we need to mention anything about setting the default of lenient in the notes here?

Reply:
Good catch, we can update the summary to mention changes to default values once it's decided what those changes are.